thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/configs/CollectionConfig.java

```diff
@@ -1,16 +1,22 @@
 package com.linkedin.thirdeye.dashboard.configs;
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 
 import com.linkedin.thirdeye.client.MetricExpression;
+import java.util.concurrent.TimeUnit;
+
 
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class CollectionConfig extends AbstractConfig {
 
   public static double DEFAULT_THRESHOLD = 0.01;
   public static String DEFAULT_TIMEZONE = "UTC";
+  public static String DEFAULT_PREAGGREGATED_KEYWORD = "all";
+  public static int DEFAULT_NONADDITIVE_BUCKET_SIZE = 5;
```

Comment:
Because the time granularity from collection schema could be different to the actual time granularity. Specifically, we have a dataset whose timestamp has time granularity in millisecond, i.e., timestamp is MILLISECOND since epoch. Hence, the time granularity in collection schema is set to "MILLISECOND" regardless its data is actually pre-aggregated in 5 MINUTES buckets.

So what would happen if we set the time granularity in collection schema to 5 MINUTES? The first problem we would encounter is that the MAX time calculation, which is computed by (timestamp x time granularity) in ThirdEye. The calculation is wrong because we are multiplying a millisecond timestamp to the 5-MINUTES granularity.

So the current work around is having a collection config to specify the real time granularity. The true solution should be changing the timestamp of the data to 5-MINUTES since epoch.